### DISCUSSION TITLE
Slow response from serverless functions with Vercel platform

### DISCUSSION THREAD

**Discussion by pepyta** [vercel/vercel#5816] [Help] [answered]
So I've deployed an app to the Vercel platform and I was wondering why it was loading slowly.

I've tested multiple options:
- Vercel with custom domain (https://ghoost.dev/api/teams/memberships/get) => 4.16s
- DigitalOcean, yet still the same source code (https://digitalocean.ghoost.dev/api/teams/memberships/get) => 400 ms

Why can it happen and how can I overcome this problem?

**Comment by pepyta** [comment, marked as answer]
As it turns out, the problem was that my Digitalocean provided database was located in Germany and Vercel server is in the USA.
This meant that every time I queried something from the database it was the communication that was slow between the two.

**Reply by aresrioja10** [reply]
Hi @pepyta ! How did you solved it?

**Reply by ErnestBrandi** [reply]
@aresrioja10 Make both host and db servers to be located close together.

**Reply by aresrioja10** [reply]
@ErnestBrandi Thanks!

**Reply by pepyta** [reply]
Vercel does not allow using selected locations for Serverless infrastructure in hobby plan, which can lead to too long response times (>10s)

If you want to use external db, you should consider switching to paid plan to select the server location or use an alternative like DigitalOcean Apps.

Using Vercel with a DigitalOcean database was a lot of pain because sometimes it worked (~600 ms response time) and sometimes it exceeded the maximally allowed 10s.

So my solution was switching to an alternative. Hope it helps!

**Comment by lorstenoplo** [comment]
@pepyta I had the same issue with serverless functions. I made a graphql api which responds like 3 times faster than the serverless approach. What do u think the problem is?
